### PR TITLE
Create feature and partner feature modules

### PR DESCRIPTION
**Pull request by eleanorreem**
- added feature module
- added partner feature module
- added minimal tests
- updated GET partner/:name endpoint to send back features. 

Note:
frontend doesn't have ID so we are stuck with using name for now

### CODE REVIEW

File: src/entities/partner-feature.entity.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  @Column()
+  partnerId: string;
+  @ManyToOne(() => PartnerEntity, (partnerEntity) => partnerEntity.partnerFeature)
+  @JoinTable({ name: 'partner', joinColumn: { name: 'partnerId' } })
+  partner: PartnerEntity;
+
+  @Column()
+  featureId: string;
+  @ManyToOne(() => FeatureEntity, (featureEntity) => featureEntity.partnerFeature, {
```
Comment (swetha-charles) [inline src/entities/partner-feature.entity.ts:21]: I might be wrong about this - but I thought the onDelete would go on the feature entity. 

Sort of similar to how the onDelete is on the partner entity for the partnerAccess relationship. I'm not a typeorm expert so I might be wrong on this

File: src/feature/feature.controller.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  constructor(private featureService: FeatureService) {}
+
+  @Post()
+  @ApiBearerAuth('access-token')
+  @ApiOperation({ description: 'Creates feature' })
+  @UseGuards(SuperAdminAuthGuard)
+  async createFeature(
+    @Body() createFeatureDto: CreateFeatureDto,
+  ): Promise<FeatureEntity | unknown> {
+    console.log('im inside create feature');
```
Comment (swetha-charles) [inline src/feature/feature.controller.ts:22]: leftover log

File: src/partner/partner.controller.ts [hunk 3]
```diff
@@ -33,10 +35,13 @@ export class PartnerController {
   }
 
   @Get(':name')
-  @ApiOperation({ description: 'Retuns profile data for a partner' })
+  @ApiOperation({ description: 'Returns profile data for a partner' })
+  @UseGuards(SuperAdminAuthGuard) // Temporary super admin auth guard
```
Comment (swetha-charles) [inline src/partner/partner.controller.ts:39]: I assume this is a temporary guard right?

**Review by swetha-charles — APPROVED** [review]

File: src/partner/partner.controller.ts (hunk 3 above)
Comment (eleanorreem) [reply src/partner/partner.controller.ts:39]: Yes

File: src/entities/partner-feature.entity.ts (hunk 1 above)
Comment (eleanorreem) [reply src/entities/partner-feature.entity.ts:21]: I think that is wrong - I watched this youtube https://www.youtube.com/watch?v=S1pxdAL2Dz8 and it seems to imply that the oncascade should be on the join table. I actually didn't put the cascade on the partner so I will do that

**Pull request merged**